### ISSUE TITLE
[FEATURE] Allow "named" versions

### ISSUE THREAD

**Issue by newhinton** [Sharkaboi/AppUpdateChecker#4]
**Is your feature request related to a problem? Please describe.**
I sometimes use version codes like this:

"v1.1.1-specialupdate"

(Those are ususally backports/special bugfixes)

As far as i can tell, this app does not support those.

**Describe the solution you'd like**
I would like to have an "allowNamedVersions" toggle, so that the name-part gets stripped from the version-check.

**Additional context**
I did try to implement this myself, since it is an rather easy change. However, build-failures are preventing me from building the library. If you want to, i can create a PR with the (untested) changes, but i could not resolve the dependency issues.

Thanks!

**Comment by Sharkaboi** [comment]
Hey

Have added support for this along with new version changes in #5 .

You can do the same using a custom version comparator mentioned [here](https://github.com/Sharkaboi/AppUpdateChecker/tree/v1.0.3?tab=readme-ov-file#custom-version-comparator). 

Will merge and release after further testing.

**Comment by newhinton** [comment]
Did you "just" write _two thousand_ changes in less than a day?!

Color me impressed. Both by the very fast nature of your help AND the fact that you "just" rewrote the entire library.

Thank you!

**Comment by Sharkaboi** [comment]
Released this with version [`v1.0.4`](https://jitpack.io/#Sharkaboi/AppUpdateChecker/v1.0.4)

**Comment by newhinton** [comment]
@Sharkaboi I am not sure why, but it seems that i cant get the latest version.
I think that there was an error in the build:

https://jitpack.io/com/github/Sharkaboi/AppUpdateChecker/v1.0.4-f52e43a2cb-1/build.log
```

Archive: com.github.sharkaboi:library:1.0.3
Publication: com.github.sharkaboi:appupdatechecker:1.0.3
Build tool exit code: 0
Looking for artifacts...
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8 -Dhttps.protocols=TLSv1.2
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8 -Dhttps.protocols=TLSv1.2
Looking for pom.xml in build directory and ~/.m2
Found artifact: com.github.sharkaboi:appupdatechecker:1.0.3
Found artifact: com.github.sharkaboi:appupdatechecker:1.0.3
Found artifact: com.github.sharkaboi:library:1.0.3
2024-03-24T14:10:35.783865776Z
Exit code: 0

✅ Build artifacts:
com.github.Sharkaboi:AppUpdateChecker:v1.0.4-f52e43a2cb-1
```
I checked with my own project, and the sample one.

**Comment by Sharkaboi** [comment]
Did a small goof and turned on minify inside the lib module, Have fixed with [v1.0.5](https://jitpack.io/#Sharkaboi/AppUpdateChecker/v1.0.5)

**Comment by newhinton** [comment]
Thanks! Now it works!
